src/ui/map.js

```diff
@@ -451,6 +451,15 @@ class Map extends Camera {
             throw new Error(`maxPitch must be less than or equal to ${defaultMaxPitch}`);
         }
 
+        // iOS 15.4 introduced a rendering bug with antialias set to `true`. Disables antialias for these devices.
+        if (!!options.antialias && window.navigator.userAgent && /\b(iPad|iPhone|iPod)\b/.test(window.navigator.userAgent)) {
+            const iosVersion = window.navigator.userAgent.match(/OS ((\d+_?){2,3})\s/);
+            if (iosVersion && iosVersion[1] === '15_4') {
+                options.antialias = false;
+                console.warn(`Antialias has been disabled for iOS 15.4 devices.`);
```

Comment:
It might make sense to use the `warnOnce` method from util.js here. I _believe_ that would prevent multiple warnings being fired (e.g. if there's multiple maps on a page).